sensor/common/centralclient/grpc_connection.go

```diff
@@ -109,9 +122,15 @@ func (f *centralConnectionFactoryImpl) SetCentralConnectionWithRetries(conn *uti
 		log.Info("Did not add central CA cert to gRPC connection")
 	}
 
-	// Use pkg/grpc configuration for MaxMsgSize on client connections as well. This will overwrite the 4MB threshold
-	// set by gRPC lib.
-	opts = append(opts, clientconn.MaxMsgReceiveSize(grpc.MaxMsgSizeSetting.IntegerSetting()))
+	var maxGRPCSize int
+	if p, err := f.getCentralGRPCPreferences(); err != nil {
+		maxGRPCSize = grpc.MaxMsgSizeSetting.IntegerSetting()
+		log.Warnf("Couldn't get gRPC preferences from central (%s). Using sensor env config (%d): %s", gRPCPreferences, maxGRPCSize, err)
+	} else {
+		maxGRPCSize = int(p.MaxGrpcReceive)
+		log.Infof("Received max gRPC size from central: %d. Overwriting sensor config", maxGRPCSize)
```

Comment:
Maybe let's get some more info out of this message?

```suggestion
		log.Infof("Received max gRPC size from central: %d. Overwriting default sensor value of %d.", maxGRPCSize, grpc.MaxMsgSizeSetting.IntegerSetting())
```